packages/client/ui/react-ui/src/hooks/useCrossmint.tsx

```diff
@@ -0,0 +1,26 @@
+import { Dispatch, SetStateAction, createContext, useContext, useState } from "react";
+
+import { Crossmint, createCrossmint } from "@crossmint/common-sdk-base";
+
+export interface CrossmintContext {
+    crossmint: Crossmint;
+    setCrossmint: Dispatch<SetStateAction<Crossmint>>;
+}
+
+const CrossmintContext = createContext<CrossmintContext | null>(null);
+
+export function CrossmintProvider({
+    children,
+    ...createCrossmintParams
+}: { children: React.ReactNode } & Parameters<typeof createCrossmint>[0]) {
+    const [crossmint, setCrossmint] = useState<Crossmint>(createCrossmint(createCrossmintParams));
```

Comment:
Discussing this on DMs -- this may cause issues if some consumer of this hook does something like...

```
const { crossmint } = useCrossmint();
const [walletSDK, setWalletSDK] = useState()<WalletsSDK>;

useEffect(() => {
  const _walletSDK = WalletSDK.from(crossmint); // this crossmint ref will be out of date later
  setWalletSDK(_walletSDK); 
}, [])

```

I think some people may choose to do this if TheirSDK.from() is a heavy initialization function. 

An additional issue is that typescript SDKs that mutate jwt, now would need to know react or take generic callback functions for updating the JWT, which seems like a lot of work

There's an alternative where we return a reference here, and observe it for updates, and emit a state change (on e.g. a 'version' property) so we maintain reactivity